src/init/chainstate.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_INIT_CHAINSTATE_H
```

Comment:
Ah I see! What do you think about putting the files under `src/node/init`? And I supposed everything would be under the `node::init` namespace?

Reply:
> Ah I see! What do you think about putting the files under `src/node/init`? And I supposed everything would be under the `node::init` namespace?

Probably best to use your own judgement, and other people may have opinions, but IMO just moving things into top level, `node`, `wallet`, `util` directories is ideal for seeing where things come from and making it obvious when you may be doing something bad like calling wallet code from node code. Extra nesting may be good, but I think only if there is some specific justification for it. In this case, I think having an "init" module is not good, because it will encourage people to add startup code from disparate parts of code there instead of closer to the actual database/index/connection structures that are being initialized. 

I also think it's nice when there's a 1:1 correspondence between directories and namespaces and personally would not suggest extra nesting for namespaces either.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/23280#discussion_r743133511

FWIW, followed up with more `node::` namespace in #/23497